internal/flow/flow.go

```diff
@@ -290,20 +296,84 @@ func PushArtifact(ctx context.Context, gitSvc *git.Service, artifactFileName, re
 	return artifactSpec.ID, nil
 }
 
-func listFiles(path string) {
+// PushArtifactToReleaseManager pushes an artifact to the release manager
+func PushArtifactToReleaseManager(ctx context.Context, releaseManagerClient *httpinternal.Client, artifactFileName, resourceRoot string) (string, error) {
+	artifactSpecPath := path.Join(resourceRoot, artifactFileName)
+	artifactSpec, err := artifact.Get(artifactSpecPath)
+	if err != nil {
+		return "", errors.WithMessagef(err, "path '%s'", artifactSpecPath)
+	}
+
+	path, err := releaseManagerClient.URLWithQuery(fmt.Sprintf("artifacts/create"), url.Values{})
+	if err != nil {
+		return "", errors.WithMessage(err, "push artifact URL generation failed")
+	}
+
+	resp := httpinternal.ArtifactUploadResponse{}
+	err = releaseManagerClient.Do(http.MethodPost, path, httpinternal.ArtifactUploadRequest{
+		Artifact: artifactSpec,
+	}, &resp)
+	if err != nil {
+		return "", errors.WithMessage(err, "create artifact request failed")
+	}
+	log.WithFields("artifactID", artifactSpec.ID, "uploadURL", resp.ArtifactUploadURL).Infof("artifact upload URL created for %s", artifactSpec.ID)
```

Comment:
It's not that critical (or obvious) but we log with the `fmt` package in methods used by the CLIs only. 
I do however think we should move away from that and configure the `log` package in CLI mode instead.

But just wanted to mention it here for awareness :)

Reply:
ok, so we can let it stay? I've seen quite a bit of difference in output, surely because logs are done in internal packages, which are then used in the CLI.

Reply:
Yes yes, let's align that in another PR

Reply:
👍